### ISSUE TITLE
ejabberd 17.04 - Captcha Generation Not Working

### ISSUE THREAD

**Issue by jwshields** [processone/ejabberd#1704]
What version of ejabberd are you using?
>ejabberd-17.04

What operating system (version) are you using?
>Debian 8.7

How did you install ejabberd (source, package, distribution)?
>`dpkg -i` 
>The .deb file from https://www.process-one.net/en/ejabberd/downloads/

What did not work as expected? Are there error messages in the log? What
was the unexpected behavior? What was the expected result?
>Result/Problem: The captcha generation does not work on the web registration page
>Expected: The captcha should generate on the web registration page

Hello all, new user to ejabberd
I've recently installed it and have been working on configuring it to my liking.
I set up the web registration and have been fiddling around with that, it seems that the captcha is not working on my system.
I'm running Debian 8.7, I have installed ImageMagick (`apt-get install imagemagick`)
I've pasted a small bit of my config, relevant to the http module and the captcha ( https://pastebin.com/gjHP3Mnc )

When I hit [host]:5280/register/new in my web browser, the page loads properly, but the captcha does not show.
Checking the error log in my jabber directory I see errors relating to the captcha script.
See below for the log,

When I am in a shell, I can call the script directly and `convert` dumps out PNG data to my terminal, so I don't believe this is a problem with ImageMagick itself, rather, the way ejabberd interprets the output from it?

Edit: I am using IM 6.8.9-9 from the Debian repo.
```
# convert -version
Version: ImageMagick 6.8.9-9 Q16 x86_64 2017-03-12 http://www.imagemagick.org
```

```
2017-04-27 07:11:46.762 [info] <0.2176.0>@ejabberd_listener:accept:311 (<0.2276.0>) Accepted connection ::FFFF:192.168.1.50:9129 -> ::FFFF:192.168.1.153:5280
2017-04-27 07:11:46.788 [debug] <0.2276.0>@ejabberd_http:init:166 S: [{[<<"websocket">>],ejabberd_http_ws},{[<<"api">>],mod_http_api},{[<<"pub">>,<<"archive">>],mod_http_fileserver},{[<<"captcha">>],ejabberd_captcha},{[<<"register">>],mod_register_web},{[<<"admin">>],ejabberd_web_admin},{[<<"http-bind">>],mod_bosh}]
2017-04-27 07:11:46.788 [info] <0.2276.0>@ejabberd_http:init:175 started: {fast_tls,{tlssock,#Port<0.45862>,#Port<0.45863>}}
2017-04-27 07:11:47.664 [info] <0.2176.0>@ejabberd_listener:accept:311 (<0.2277.0>) Accepted connection ::FFFF:192.168.1.50:9130 -> ::FFFF:192.168.1.153:5280
2017-04-27 07:11:47.664 [debug] <0.2277.0>@ejabberd_http:init:166 S: [{[<<"websocket">>],ejabberd_http_ws},{[<<"api">>],mod_http_api},{[<<"pub">>,<<"archive">>],mod_http_fileserver},{[<<"captcha">>],ejabberd_captcha},{[<<"register">>],mod_register_web},{[<<"admin">>],ejabberd_web_admin},{[<<"http-bind">>],mod_bosh}]
2017-04-27 07:11:47.664 [info] <0.2277.0>@ejabberd_http:init:175 started: {fast_tls,{tlssock,#Port<0.45864>,#Port<0.45865>}}
2017-04-27 07:11:47.676 [debug] <0.2277.0>@ejabberd_http:process_header:300 ({tlssock,#Port<0.45864>,#Port<0.45865>}) http query: 'GET' <<"/register/new">>
2017-04-27 07:11:47.676 [debug] <0.2277.0>@ejabberd_http:process:387 [<<"register">>,<<"new">>] matches [<<"register">>]
2017-04-27 07:11:47.717 [error] <0.2277.0>@ejabberd_captcha:do_create_image:336 Failed to process output from "/opt/ejabberd-17.04/lib/ejabberd-17.04/priv/bin/captcha.sh 111294". Maybe ImageMagick's Convert program is not installed.
```

```
2017-04-27 07:11:47.717 [error] <0.2277.0>@ejabberd_captcha:do_create_image:336 Failed to process output from "/opt/ejabberd-17.04/lib/ejabberd-17.04/priv/bin/captcha.sh 111294". Maybe ImageMagick's Convert program is not installed.
```

**Comment by badlop** [comment]
I can reproduce the problem. Surprisingly, if I install my ejabberd, and configure it to use that path, then it starts correctly.

I supected it was a permission problem, because in other cases I saw:
> sh: 1: exec: /opt/ejabberd-17.04/lib/ejabberd-17.04/priv/bin/captcha.sh: Permission denied
> 13:09:20.700 [error] Failed to process output from
> "/opt/ejabberd-17.04/lib/ejabberd-17.04/priv/bin/captcha.sh 465862".
> Maybe ImageMagick's Convert program is not installed.

So I granted permission to execute captcha.sh to everybody, but the problem persists.

**Comment by jwshields** [comment]
@badlop Thanks for your comment. My script was set with permissions of 750 and owned by the uid 1003 (which is not a "real"/valid user on my system) - I assume this is something left over or broken from when the deb installer was created?

I set the perms of that file to 777 (I know that's bad, I've changed it back), and the captcha started working. After that is when I noticed the owner, so I've changed the perms back to 750, and did a `chown -R ejabberd *` on the whole directory where I've installed ejabberd. 
Now the captcha is generating every time I hit the page.

I want to say this issue can be closed, but I feel like a fix might need to be employed to the current package.
